base/src/org/compiere/model/MChartDatasource.java

```diff
@@ -176,20 +198,29 @@ else if ( parent.getTimeUnit().equals(MChart.TIMEUNIT_Quarter))
 					else if ( parent.getTimeUnit().equals(MChart.TIMEUNIT_Year))
 						period = new Year(date);
 
-					tseries.add(period, rs.getBigDecimal(1));
-					key = period.toString();
+					if (period != null) {
```

Comment:
@EdwinBetanc0urt  can please, use the funcional way to management,  Opcional maybeKey(period.toString()) or  Opcional maybePeriod(period.toString())